### ISSUE TITLE
Document type cache not thread-safe.

### ISSUE THREAD

**Issue by KorsG** [UblSharp/UblSharp#20]
I process multiple UblDocuments concurrently and sometimes get a NullReferenceException while loading documents via the "Load" method in the "UblDocument" class.

I traced it to the "GetSerializer" method since it uses a Dictionary without a lock.
_(https://github.com/UblSharp/UblSharp/blob/master/src/UblSharp/UblDocument.cs#L76)_

I would have created a pull request, but I can't get the project to build, so I am just going to post the fix here:

```csharp
        public static XmlSerializer GetSerializer(Type type)
        {
            XmlSerializer serializer;
            // Lock the type cache dictionary in case method is accessed concurrently, since a dictionary is not thread-safe when both reading and writing.
            lock (s_typeCache)
            {
                if (!s_typeCache.TryGetValue(type, out serializer))
                {
                    serializer = new XmlSerializer(type);
                    s_typeCache[type] = serializer;
                }
            }

            return serializer;
        }
```

PS. thank you for a great project! :-)

**Comment by remcoros** [comment]
Good catch, I'll fix this soon using ConcurrentDictionary probably.

**Comment by remcoros** [comment]
Fixed in 1.1.1
